scripts/pages/zoning/zoning.ts

```diff
@@ -559,21 +559,37 @@ class ZoneMenuHandler {
 
 	deleteRegion() {
 		if (this.selectedZone.zone?.regions?.length === 1) {
-			$.Warning("Deleted the last Region in this zone!!!");
-			const trackIndex =
-				this.selectedZone.track === this.mapZoneData.tracks.main
-					? 0
-					: this.mapZoneData.tracks.bonuses.indexOf(this.selectedZone.track) + 1;
-			const segmentIndex = this.selectedZone.track.zones.segments.indexOf(this.selectedZone.segment);
-
-			this.deleteSelection();
-			const childContainer = this.panels.trackList.GetChild(trackIndex)?.FindChildTraverse('ChildContainer');
-			const segmentPanel = childContainer?.FindChildTraverse('SegmentContainer')?.GetChild(segmentIndex);
-			const selectButton = segmentPanel.FindChildTraverse<ToggleButton>('SelectButton');
-			const deleteButton = segmentPanel.FindChildTraverse<Button>('DeleteButton');
-
-			selectButton?.SetSelected(true);
-			this.updateSelection(this.selectedZone.track, this.selectedZone.segment, null, deleteButton);
+			UiToolkitAPI.ShowGenericPopupTwoOptionsBgStyle(
+				$.Localize('#Zoning_Delete'),
+				$.Localize('#Zoning_LastRegion_Message'),
+				'warning-popup',
+				$.Localize('#Zoning_Delete'),
+				() => {
+					const trackIndex =
```

Comment:
I would refactor this into a reusable function if this code is repeated more than twice.

Reply:
I can't find another repeat of this code (but it's entirely possible I'm just overlooking how this would be generalized for another similar use)

Reply:
Yeah unless there's genuine repeats of this, generally I avoid refactoring something out to a separate method just because it's in an arrow function. Inline arrow functions like this are very readable!